### ISSUE TITLE
Cannot convert a Symbol value to a string

### ISSUE THREAD

**Issue by unional** [Microsoft/TypeScript#12169]
tsc: 2.1.0-dev.20161110

```ts
function foo(x: Symbol) {
    console.log(`abc ${x}`)
}
```

transpile to:
```js
function foo(x) {
    console.log("abc " + x);
}
```

results in "Cannot convert a Symbol value to a string" error at runtime.

http://www.typescriptlang.org/play/#src=function%20foo(x%3A%20Symbol)%20%7B%0D%0A%20%20%20%20console.log(%60abc%20%24%7Bx%7D%60)%0D%0A%7D

**Comment by Kovensky** [comment]
`Symbol` values indeed can't be converted to any other type (other than by calling `.toString()`), and the type checker does catch other match-ups such as with `number` or `Object`, but it allows `string`...

**Comment by unional** [comment]
Yes, but in this case, as it is used in literal string, I would suggest to transpile it to `.toString()`:

```js
function foo(x) {
  console.log("abc " + x.toString())
}
```

The question is what to do when `x: Symbol | SomeOtherType`.

**Comment by blakeembrey** [comment]
@unional I don't think TypeScript should change the function call implicitly. If anything, I think this should generate a TypeScript error for non-stringable entities like `symbol`.

Side note: What value is `Symbol`? I only know of `symbol`.

**Comment by unional** [comment]
I can see that it may not make sense to transform it to `symbol.toString()`, as described here:

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals
> The default function just concatenates the parts into a single string.

And just tried on the chrome devTools and indeed `var a = Symbol(); \`abc ${a}\`` yield the same error.

By the way, is there any other "non-stringable" entities?

**Comment by blakeembrey** [comment]
> By the way, is there any other "non-stringable" entities?

Not much comes to mind, but there's `Object.create(null)`.

**Comment by unional** [comment]
If that's the case, would it be an overlooked case on JavaScript template literal side? Because:

```js
var a = Symbol('a');
console.log(a); // "Symbol(a)"
console.log(`abc ${a}`); // TypeError: Cannot convert a Symbol value to a string
```

**Comment by blakeembrey** [comment]
No, you can do `String(Symbol('a'))` to convert it to a string. JavaScript's coercion rules can be a little confusing.

**Comment by unional** [comment]
Yeah, I re-read the doc and here it is:

> Symbol("foo") + "bar" throws a TypeError (can't convert symbol to string). This prevents you from silently creating a new string property name from a symbol, for example.
The "safer" String(sym) conversion works like a call to Symbol.prototype.toString() with symbols, but note that new String(sym) will throw.

```js
var a = Symbol('a');
`abc ${String(a)}` // "abc Symbol(a)"
`abc $(new String(a)}` // TypeError: Cannot convert a Symbol value to a string
```

😞
